src/toolbox/setup/windows.ts

```diff
@@ -1,5 +1,190 @@
+        if (x === BIN_PATH)
+          found = true
+  
+      if (found) {
+        return {success: true, info: "Moddable BIN already in User PATH."}
+      } else {
+        const value = {
+          "HKCU\\ENVIRONMENT": {
+            "Path": {
+              value: pathString + ";" + BIN_PATH,
+              type: "REG_EXPAND_SZ"
+            }
+          }
+        }
+  
+        await regedit.promisified.putValue(value)
+        return {success: true}
+      }
+    }
+  } catch (error) {
+    return {success: false, info: `Error setting registry value`}
+  }
+  return {success: false}
+}
 
 export default async function (): Promise<void> {
-  print.warning('Windows setup is not currently supported')
+  const BIN_PATH = filesystem.resolve(
+    INSTALL_PATH,
+    'build',
+    'bin',
+    'win',
+    'release'
+  )
+  const BUILD_DIR = filesystem.resolve(
+    INSTALL_PATH,
+    'build',
+    'makefiles',
+    'win'
+  )
+
+  print.info(`Setting up Windows tools at ${INSTALL_PATH}`)
+
+  // Check for Visual Studio CMD tools
+  try {
+    await system.exec('nmake /?')
+  } catch (error) {
+    print.error('Visual Studio 2022 Community is required to build the Moddable SDK: https://www.visualstudio.com/downloads/')
+    // @@TODO: need to check if perhaps VS.bat has not been run...
+    process.exit(1)
+  }
+  try {
+    await system.exec(`git --version`)
+  } catch (error) {
+    print.error('git is required to clone the Moddable SDK: https://git-scm.com/download/win')
+    process.exit(1)
+  }
+
+  const spinner = print.spin()
+  spinner.start('Beginning setup...')
+
+  // 1. clone moddable repo into INSTALL_DIR directory if it does not exist yet
+  try {
+    filesystem.dir(INSTALL_DIR)
+
+    await upsert(EXPORTS_FILE_PATH, `# Generated by xs-dev CLI`)
+  } catch (error) {
+    spinner.fail(`Error setting up install directory: ${String(error)}`)
+    process.exit(1)
+  }
+
+  if (filesystem.exists(INSTALL_PATH) !== false) {
+    spinner.info('Moddable repo already installed')
+  } else {
+    try {
+      spinner.start('Cloning Moddable-OpenSource/moddable repo')
+      await system.spawn(`git clone ${MODDABLE_REPO} ${INSTALL_PATH}`)
+      spinner.succeed()
+    } catch (error) {
+      spinner.fail(`Error cloning moddable repo: ${String(error)}`)
+      process.exit(1)
+    }
+  }
+
+  // 2. configure MODDABLE env variable, add release binaries dir to PATH
+  
+  spinner.start(`Adding Moddable SDK to User Environment`)
+  let result = await setEnv("MODDABLE", INSTALL_PATH, RegType.REG_SZ)
+  if (result.success) {
+    if (result.info)
+      spinner.info(result.info)
+    spinner.succeed()
+  } else {
+    spinner.fail(result.info)
+  }
+  process.env.MODDABLE = INSTALL_PATH
+
+  spinner.start(`Adding Moddable SDK to User Path`)
+  result = await addToPath(BIN_PATH)
+  if (result.success) {
+    if (result.info)
+      spinner.info(result.info)
+    spinner.succeed()
+  } else {
+    spinner.fail(result.info)
+  }
```

Comment:
With the suggested refactor to `addToPath`:

```suggestion
  try {
    const result = await addToPath(BIN_PATH)
    spinner.info(result)
    spinner.succeed()
  } catch (error) {
    spinner.fail(error.toString())
  }
```